crates/vault-registry/src/types.rs

```diff
@@ -147,7 +149,7 @@ impl Default for VaultStatus {
 }
 
 #[derive(Encode, Decode, Clone, PartialEq)]
-#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "std", derive(Debug, serde::Serialize))]
```

Comment:
Do we also need to derive `Deserialize`?

Reply:
I'll check if the client requires it

Reply:
It does